src/DIRAC/Core/DISET/private/BaseClient.py

```diff
@@ -70,8 +70,8 @@ def __init__(self, serviceName, **kwargs):
     if not isinstance(serviceName, six.string_types):
       raise TypeError("Service name expected to be a string. Received %s type %s" %
                       (str(serviceName), type(serviceName)))
-    self._destinationSrv = serviceName
-    self._serviceName = serviceName
+    self._destinationSrv = str(serviceName)
```

Comment:
Please add a comment that this is for python3 compatibility. 
One day, we may want to cleanup...

Reply:
As the issue is Python 2 specific and has nothing to do with Python 3:
```suggestion
    # Explicitly convert to a str to avoid Python 2 M2Crypto issues with unicode objects
    self._destinationSrv = str(serviceName)
```